src/Apps/Order/Components/Validators.tsx

```diff
@@ -0,0 +1,21 @@
+import { reduce } from "lodash"
+
+// Object.assign() because spread operators are not supported on generic types at the moment:
+// https://stackoverflow.com/questions/45384697/why-does-typescript-complain-that-an-object-must-be-an-object-in-spread-types
+export const requiredFields: <
+  T extends { [propName: string]: string },
+  K extends keyof T,
+  E extends Partial<T>
+>(
+  fields: K[],
+  values: T,
+  startingErrors?: E
+) => E = (fields, values, startingErrors = {} as any) =>
+  reduce(
+    fields,
+    (acc, fieldName) =>
+      values[fieldName].length
+        ? acc
+        : Object.assign({}, acc, { [fieldName]: "Required" }),
+    startingErrors
+  )
```

Comment:
Since there is a lot of data to unpack here (which in the future might need to be `console.log`'d) can you add brackets to each function? Implicit returns should never be used on code that exceeds one line as it makes inspectability more difficult as well as introduces the potential for subtle errors over time.